src/parsers/manifest/dash/parse_mpd.ts

```diff
@@ -274,21 +274,29 @@ function parseCompleteIntermediateRepresentation(
     minimumTime = contentStart !== undefined            ? contentStart :
                   parsedPeriods[0]?.start !== undefined ? parsedPeriods[0].start :
                                                           0;
-    let maximumTime : number | undefined;
-    if (contentEnd !== undefined) {
-      maximumTime = contentEnd;
-    } else if (mediaPresentationDuration !== undefined) {
+    let maximumTime = Infinity;
+    if (mediaPresentationDuration !== undefined &&
+        mediaPresentationDuration < maximumTime) {
```

Comment:
That condition should be always true considering that `maximumTime` has just been set to `Infinity`.

Maybe we should just initialize `maximumTime` to `mediaPresentationDuration ?? Infinity`